fixtures/asset-config/test-cases/encoding-test-cases.ts

```diff
@@ -0,0 +1,335 @@
+import { TestCase } from "../html-handling.test";
+
+export const encodingTestCases: {
+	html_handling:
+		| "auto-trailing-slash"
+		| "drop-trailing-slash"
+		| "force-trailing-slash"
+		| "none";
+	cases: TestCase[];
+}[] = [
+	{
+		html_handling: "auto-trailing-slash",
+		cases: [
+			{
+				title: "/[boop] -> /%5Bboop%5D 307 (with /[boop].html)",
+				files: ["/[boop].html"],
+				requestPath: "/[boop]",
+				matchedFile: "/[boop].html",
+				finalPath: "/%5Bboop%5D",
+			},
+			{
+				title: "/%5Bboop%5D -> 200 (with /[boop].html)",
+				files: ["/[boop].html"],
+				requestPath: "/%5Bboop%5D",
+				matchedFile: "/[boop].html",
+				finalPath: "/%5Bboop%5D",
+			},
+			{
+				title: "/%5Bboop%5D -> 200 (with /%5Bboop%5D.html)",
+				files: ["/%5Bboop%5D.html"],
+				requestPath: "/%5Bboop%5D",
+				matchedFile: "/%5Bboop%5D.html",
+				finalPath: "/%5Bboop%5D",
+			},
+			{
+				title: "/[boop] -> 307 (with /%5Bboop%5D.html)",
+				files: ["/%5Bboop%5D.html"],
+				requestPath: "/[boop]",
+				matchedFile: "/%5Bboop%5D.html",
+				finalPath: "/%5Bboop%5D",
+			},
+			{
+				title: "/%5Bboop%5D.html -> 307 (with /%5Bboop%5D.html)",
+				files: ["/%5Bboop%5D.html"],
+				requestPath: "/%5Bboop%5D.html",
+				matchedFile: "/%5Bboop%5D.html",
+				finalPath: "/%5Bboop%5D",
+			},
+			// mix of encoded and unencoded paths
+			{
+				title: "/beep/[b%C3%B2op] -> /beep/%5Bb%C3%B2op%5D 307",
+				files: ["/beep/[bòop].html"],
+				requestPath: "/beep/[b%C3%B2op]",
+				matchedFile: "/beep/[bòop].html",
+				finalPath: "/beep/%5Bb%C3%B2op%5D",
+			},
+			// exact matches should win
+			{
+				title: "/[boop] -> 200 (with /[boop].html)",
+				files: ["/%5Bboop%5D.html", "/[boop].html"],
+				requestPath: "/[boop]",
+				matchedFile: "/[boop].html",
+				finalPath: "/[boop]",
+			},
+			{
+				title: "/%5Bboop%5D -> 200 (with /%5Bboop%5D.html)",
+				files: ["/%5Bboop%5D.html", "/[boop].html"],
+				requestPath: "/%5Bboop%5D",
+				matchedFile: "/%5Bboop%5D.html",
+				finalPath: "/%5Bboop%5D",
+			},
```

Comment:
Square brackets can be not encoded and so both versions could be uploaded and accessed. However, a user could upload two versions of something with characters that must be encoded, would not be able to access the unencoded file name.
e.g. a user tries to upload /beep/bòop.html and /beep/b%C3%B2op.html. Only /beep/b%C3%B2op.html can ever be accessed

Reply:
Similarly, `/beep/bòop.html` would be available at `/beep/b%C3%B2op` and `/beep/b%C3%B2op.html` would be available at `/beep/b%25C3%25B2op`. Totally daft, but they're being weird by uploading us encoded paths, so they asked for it.